### ISSUE TITLE
Errors with Tf_results

### ISSUE THREAD

**Issue by rubyruins** [flow2ml/Flow2ML#143]
### **Describe the bug:**
According to the readme, to use tf_results we must pass the model and a validation generator to it. However, there seems to be some error that prevents it from working correctly.

### **To Reproduce:**
I've created a simple model with a very small dataset using datagenerator. Passing the model and validation generator to the function raises many exceptions:

```
Traceback (most recent call last):
  File ".\temp.py", line 48, in <module>
    x.tf_get_results_docx()
  File "C:\Users\ACER\Downloads\Flow2ML\flow2ml\Tf_Results.py", line 95, in tf_get_results_docx
    self.tf_roc_curve()
  File "C:\Users\ACER\Downloads\Flow2ML\flow2ml\Tf_Results.py", line 37, in tf_roc_curve
    fpr, tpr, _ = roc_curve(labels, preds)
  File "C:\Users\ACER\Downloads\Flow2ML\venv\lib\site-packages\sklearn\utils\validation.py", lier_f
    return f(*args, **kwargs)
  File "C:\Users\ACER\Downloads\Flow2ML\venv\lib\site-packages\sklearn\metrics\_ranking.py", lic_curve
    fps, tps, thresholds = _binary_clf_curve(
  File "C:\Users\ACER\Downloads\Flow2ML\venv\lib\site-packages\sklearn\metrics\_ranking.py", liinary_clf_curve
    y_score = column_or_1d(y_score)
  File "C:\Users\ACER\Downloads\Flow2ML\venv\lib\site-packages\sklearn\utils\validation.py", lier_f
    return f(*args, **kwargs)
  File "C:\Users\ACER\Downloads\Flow2ML\venv\lib\site-packages\sklearn\utils\validation.py", lilumn_or_1d
    raise ValueError(
ValueError: y should be a 1d array, got an array of shape (4, 2) instead.
```

The model has been successfully trained and I can see its progress with every epoch. The errors only start when trying calling the function as follows:
```
x = Tf_Results(model,val_generator)
x.tf_get_results_docx() 
```

**Comment by chebroluharika** [comment]
@ArkaprabhaChakraborty can you check this. This is important to fix this before program ends

**Comment by ArkaprabhaChakraborty** [comment]
can I see the validation generator?

**Comment by rubyruins** [comment]
Yep, here it is. Let me know if I should make changes to it:

```
datagen = ImageDataGenerator(validation_split = 0.25, rescale = 1./255)

train_generator = datagen.flow_from_directory(
    './dataset_dir/data_dir', 
    subset = 'training',
    target_size = (300,300),
    batch_size = 2,
    class_mode = 'categorical',
)

val_generator = datagen.flow_from_directory(
    './dataset_dir/data_dir', 
    subset = 'validation',
    target_size = (300,300),
    batch_size = 2,
    class_mode = 'categorical'
)
```

**Comment by rubyruins** [comment]
Yep, here it is. Let me know if I should make changes to it:

```
datagen = ImageDataGenerator(validation_split = 0.25, rescale = 1./255)

train_generator = datagen.flow_from_directory(
    './dataset_dir/data_dir', 
    subset = 'training',
    target_size = (300,300),
    batch_size = 16,
    class_mode = 'categorical',
)

val_generator = datagen.flow_from_directory(
    './dataset_dir/data_dir', 
    subset = 'validation',
    target_size = (300,300),
    batch_size = 16,
    class_mode = 'categorical'
)
```

**Comment by ArkaprabhaChakraborty** [comment]
Alright, I'm facing this error :)
![Screenshot from 2021-05-30 22-07-58](https://user-images.githubusercontent.com/58482073/120112840-377a9380-c195-11eb-81c5-f560e11301da.png)

**Comment by ArkaprabhaChakraborty** [comment]
So I decided to test out Tf_Results.py script. And I got this:
![download](https://user-images.githubusercontent.com/58482073/120112889-7c9ec580-c195-11eb-9b51-f4f6caa1e7b9.png)
![download (1)](https://user-images.githubusercontent.com/58482073/120112890-7f011f80-c195-11eb-9f09-cbe61f791fc5.png)
![download (2)](https://user-images.githubusercontent.com/58482073/120112891-80cae300-c195-11eb-9e0b-f563d60d7648.png)
So it's working fine. Now, what is causing the error on your end is that validation generator class labels isn't of the same shape as that of the model's prediction.
